sips/sip-02X/sip-02X-wallet-interface.md

```diff
@@ -0,0 +1,500 @@
+However, wallets implementing the new standard are advised to maintain the previous system to support legacy applications during a transition period.
+Existing applications using the current Auth system should continue to operate, but immediate changes are recommended once this SIP is ratified.
+The Connect library should be go live with the new standard, once most major wallets have adopted the new standard.
+
+# Implementation
+
+## Notes on Serialization
+
+The WebBTC request standard is inspired by JSON-RPC 2.0 for serialization.
+To adhere to a generic serializability, the following notes are given.
+Enums are serialized as humand-readable strings.
+BigInts are serialized as numbers, strings, or anything that can be parsed by the JavaScript BigInt constructor.
+Bytes are serialized as hex-encoded strings (without a 0x prefix).
+Predefined formats from previous SIPs are used where applicable.
+Addresses are serialized as Stacks c32-encoded strings.
+Clarity values, post-conditions, and transactions are serialized to bytes (defined by SIP-005) and used as hex-encoded strings.
+
+## Methods
+
+This section defines the available methods, their parameters, and result structure.
+Parameters should be considered recommendations for the wallet.
+The user/wallet may choose to ignore/override them.
+Optional params are marked with a `?`.
+
+Methods can be namespaced under `stx_` if used in settings like WebBTC (see WBIP-002) and other more Ethereum inspired domains.
+In other cases (e.g. WalletConnect), the namespace may already be given by meta-data (e.g. a `chainId` field) and can be omitted.
+On the predominant `StacksProvider` global object, the methods can be used without a namespace, but wallets may add namespaced aliases for convenience.
+
+#### Common definitions
+
+The following definitions can be used in multiple methods (mainly for transfer and transaction methods).
+
+`params`
+
+- `address?`: `string` address, Stacks c32-encoded, defaults to wallets current address
+- `network?`: `'mainnet' | 'testnet' | 'regtest' | 'mocknet'`
+- `fee?`: `number | string` BigInt constructor compatible value
+- `nonce?`: `number | string` BigInt constructor compatible value
+- `attachment?`: `string` hex-encoded
+- `anchorMode?`: `'on-chain' | 'off-chain' | 'any'`
+- `postConditions?`: `PostCondition[]`, defaults to `[]`
+- `postConditionMode?`: `'allow' | 'deny'`
+- `sponsored?`: `boolean`, defaults to `false`
+- ~~`appDetails`~~ _removed_
+- ~~`onFinish`~~ _removed_
+- ~~`onCancel`~~ _removed_
+
+`where`
+
+- `PostCondition`: `string | object` hex-encoded or JSON representation
+
+---
+
+### Method `stx_transferStx`
+
+> **Comment**: This method doesn't take post-conditions.
+
+`params`
+
+- `recipient`: `string` address, Stacks c32-encoded
+- `amount`: `number | string` BigInt constructor compatible value
+- `memo?`: `string`, defaults to `''`
+
+`result`
+
+- `txid`: `string` hex-encoded
```

Comment:
Seems that many of the param names in this spec are camelCased, so doesn't `txid` stand out as not following this general pattern? If we're already using a `transaction` parameter in many calls, why not use `transactionId` here? It would keep in line with the camelCasing, as well as JS's generally verbose names. 

From what I've seen, codebases dealing with transactions use both `transaction` and `tx` in variable names (and so does this spec!), so I'd propose sticking with `transaction`, and help maintain the use of `transaction` in various related specs by updating the [WBIP003 draft](https://webbtc.netlify.app/wbips/WBIP003) to include a case for the term `transaction`. Wdyt?

Reply:
We had this discussion a while back, it was `txId` before, but since txid is a term on its own in the bitcoin/blockchain world, we went with `txid`

Reply:
I guess where I was coming from with the suggestion is that programming environments do their best to refer to the same ideas using their own conventions. For a transaction ID, I'd assume that `txid` would be more in line with how variables are named in lower level languages. And so in this spec it kind of stands out for not using camelCase as is more typical of JS environments, plus we're already using the word `transaction` in other areas :sweat_smile:

Reply:
I get that. Both make sense to me. `txid` is a bit more Bitcoin aligned. It's referred to as a word/concept/thing in the core node etc. https://github.com/search?q=repo%3Astacks-network%2Fstacks-core%20txid&type=code so I feel it's what developers expect.